common/journald/src/journalctl/mod.rs

```diff
@@ -0,0 +1,433 @@
+         field name, followed by '=', followed by field data, followed by a newline as separator to the next field.
+     Other journal fields are serialized in a special binary safe way:
+         field name, followed by newline, followed by a binary 64bit little endian size value, followed by the binary field data,
+         followed by a newline as separator to the next field.
+     Entry metadata that is not actually a field is serialized like it was a field, but beginning with two underscores.
+     More specifically, __CURSOR=, __REALTIME_TIMESTAMP=, __MONOTONIC_TIMESTAMP= are introduced this way.
+     The order in which fields appear in an entry is undefined and might be different for each entry that is serialized.
+
+     And that's it.
+     */
+
+    any_partial_state(
+        (
+            many1::<std::collections::HashMap<_, _>, _, _>(
+                (
+                    // Get the key
+                    recognize(skip_many1(choice((letter(), token(b'_'))))).and_then(
+                        |bytes: &[u8]| {
+                            std::str::from_utf8(bytes)
+                                .map(|s| s.to_string())
+                                .map_err(StreamErrorFor::<Input>::other)
+                        },
+                    ),
+                    // Get the Value
+                    one_of(b"=\n"[..].iter().copied()).then_partial(|&mut c| {
+                        // If the k/v separator is a = it's just utf8
+                        if c == b'=' {
+                            recognize(skip_many(none_of(b"\n".iter().copied())))
+                                .and_then(|bytes: &[u8]| {
+                                    std::str::from_utf8(bytes)
+                                        .map(|s| FieldValue::Utf8(s.to_string()))
+                                        .map_err(StreamErrorFor::<Input>::other)
+                                })
+                                .left()
+                        } else {
+                            // else it must be a newline, therefor treat it as bytes
+                            le_u64()
+                                .then_partial(|vlen: &mut u64| {
+                                    let vlen: usize = (*vlen).try_into().unwrap();
+                                    take(vlen)
+                                        .map(|bytes: &[u8]| FieldValue::Bytes(bytes.to_owned()))
+                                })
+                                .right()
+                        }
+                    }),
+                    // lines within an entry are line separated
+                    token(b'\n'),
+                )
+                    .map(|(key, value, _)| (key, value)),
+            ),
+            // entries are line separated
+            token(b'\n'),
+        )
+            .map(|(kvs, _)| kvs),
+    )
+}
+
+impl JournaldExportDecoder {
```

Comment:
This is unfortunately alloc'y I think I know how to get it working with the client's lazy lines support, but it pollutes the decoder logic a bit so it's probably best left until later.